### ISSUE TITLE
Left option key as meta

### ISSUE THREAD

**Issue by leezu** [jgosmann/neo2-layout-osx#8]
Thanks @jgosmann for your work on the neo layout files for OS X.
Unfortunately the layout maps the left and right option keys to MOD3, whereas "usually" only Capslock should be MOD3. I assume this is due to mapping Capslock to MOD3 directly being impossible?

With Karabiner Elements it is possible to map Capslock to MOD3 (by mapping it to left_option, which in turn thanks to the neo.keylayout is MOD3). Unfortunately this doesn't release the left option key to it's original use. I would like to use the left option key as Meta key for use with Emacs.

To get back the left option key, I tried replacing `anyOption` with `rightOption` in the layout definition. Unfortunately this did not have the intended effect and the both left and right option keys stopped having any effect.
Do you have any suggestions about a way to go forward?

**Comment by jgosmann** [comment]
The way the layout is designed MOD3 is fundamentally the same thing as `option`. Most software will also not differentiate between `left_option` and `right_option`. Your best bet might be remapping the option keys to some other key that you can use as meta key in Karabiner Elements. But I don't know which could you might be able to use for that ...

**Comment by leezu** [comment]
Thanks @jgosmann. Then my distinction between left and right option is not sensible. Is there any reason why you don't use the

**Comment by leezu** [comment]
Thanks @jgosmann. Then my distinction between left and right option is not sensible. Is there any reason why not the `caps` key is used to represent MOD3?

**Comment by jgosmann** [comment]
A couple of reasons:
1. The original keyboard layout file that I took to improve did it this way. Changing it could be quite a bit of work that I currently don't want to invest.
2. I'm not an expert, but my understanding is that keyboard layout layers are defined with respect to the modifier keys that activate the layer. These modifier keys are option, shift, command, control, caps lock. There is no MOD3, so the layer three has to be activated by a set of these modifiers.
3. The caps lock layer could be used for layer 3, but then there would be no caps lock layer which is also useful to have.
4. The option key is usually used to enter layer 3 of a keyboard layout in macOS (i.e., even in regular keyboard layouts it is used to type special symbols).

It might be possible to use Karabiner Elements to implement layer 3 similar to layer 4, but this probably comes with its own set of problems as then every keypress will be converted to a sequence of keys (activating a dead key to tell the layout to produce a layer 3 symbol and then another key producing the symbol).

**Comment by leezu** [comment]
Thanks @jgosmann for the elaboration. Fortunately I found a workaround for my problem based on the following observation

- neo.keylayout defines the keymap using `anyOption` as MOD3
- Karabiner Elements maps caps lock to `leftOption`
- A running application can do further key mapping. In particular GUI Emacs allows to separately set the action of left / right option, command, etc. keys. Also terminal emulators (at least iterm2) can do such remapping.
- The choice of Karabiner Elements is arbitrary and with no other changes mapping `caps_lock`→`rightOption` is equivalent to the current user visible behavior.

However, changing the Karabiner Elements mapping for caps lock from `caps_lock`→`leftOption` to `caps_lock`→`rightOption` makes it possible to overwrite `leftOption` in any application (ie. GUI Emacs / terminal). Previously, overriding `leftOption` would mean that also `caps_lock` was overwritten, as both keys are equivalent thanks to Karabiner. Obviously now the same applies to `rightOption`, but personally I don't need to use `rightOption`.

In particular I made the following changes:

1. Change  `{"from":{"key_code":"caps_lock","modifiers":{"optional":["any"]}},"to":[{"key_code":"left_option"}],"type":"basic"}` to `{"from":{"key_code":"caps_lock","modifiers":{"optional":["any"]}},"to":[{"key_code":"right_option"}],"type":"basic"}` in `~/.config/karabiner/karabiner.json`.
2. Add   `(setq mac-option-modifier 'meta) ; make option key do Meta` and `(setq mac-right-option-modifier nil) ; disable right opt key (for neo2)` to the emacs init file (for GUI Emacs).
3. Set "Left option" sends "Esc+" in iterm2 preferences → Profile → Keys
